tracer_test.go

```diff
@@ -317,6 +317,48 @@ func TestGen128Bit(t *testing.T) {
 	assert.True(t, traceID.Low != 0)
 }
 
+func TestHighTraceIDGenerator(t *testing.T) {
+	id := uint64(12345)
+	calledGenerator := false
+	highTraceIDGenerator := func() uint64 {
+		calledGenerator = true
+		return id
+	}
+
+	tracer, tc := NewTracer("x", NewConstSampler(true), NewNullReporter(), TracerOptions.HighTraceIDGenerator(highTraceIDGenerator), TracerOptions.Gen128Bit(true))
+	defer tc.Close()
+
+	span := tracer.StartSpan("test", opentracing.ChildOf(emptyContext))
+	defer span.Finish()
+	traceID := span.Context().(SpanContext).TraceID()
+	assert.Equal(t, id, traceID.High)
+	assert.True(t, calledGenerator)
+}
+
+type mockLogger struct {
```

Comment:
shouldn't do that. We have log/zap package where you can create a logger with zap, and zap has a ByteBuffer-backed logger. To make using it easier, however, we can add log/testutils package that will do the above in one call.

Reply:
incidentally, we have something like that in the main repo, `jaeger/pkg/testutils/logger.go`, another option would be to consolidate that in jaeger-lib (but I'd punt on that for now)

Reply:
When I try using a the `log/zap` package in my test case, I get this error:

```
# github.com/uber/jaeger-client-go
import cycle not allowed in test
package github.com/uber/jaeger-client-go (test)
	imports github.com/uber/jaeger-client-go/log/zap
	imports github.com/uber/jaeger-client-go
```

Imagine this needs to be moved to jaeger-lib or otherwise refactored (the issue is `log/zap/field.go` not `log/zap/logger.go`).

Reply:
Should I just copy the struct from `log/zap/logger.go` for now or should I refactor `testutils` into `jaeger-lib` as you mentioned above?

Reply:
just leave it for now, add a TODO